src/main/java/org/cbioportal/web/columnar/StudyViewColumnStoreController.java

```diff
@@ -362,11 +362,19 @@ public ResponseEntity<ClinicalViolinPlotData> fetchClinicalDataViolinPlots(
         List<ClinicalData> patientClinicalDataList = filterNonEmptyClinicalData(
             studyViewColumnarService.getPatientClinicalData(interceptedStudyViewFilter, attributeIds)
         );
-
-        List<ClinicalData> combinedClinicalDataList = Stream.concat(
-            sampleClinicalDataList.stream(),
-            convertPatientClinicalDataToSampleClinicalData(patientClinicalDataList, filteredSamples).stream()
-        ).toList();
+        
+        List<ClinicalData> combinedClinicalDataList;
+        if (patientClinicalDataList.isEmpty()) {
+            combinedClinicalDataList = sampleClinicalDataList;
+        } else {
+            // fetch samples again without the numerical clinical data filter
```

Comment:
@onursumer can you add to comment WHY we fetch again without clinical data filter?

Reply:
done!